server/config_push_proxy.go

```diff
@@ -12,16 +12,21 @@ import (
 )
 
 type ConfigPushProxy struct {
-	AndroidPushSettings     []AndroidPushSettings
-	ListenAddress           string
-	ThrottleVaryByHeader    string
-	LogFileLocation         string
-	SendTimeoutSec          int
-	RetryTimeoutSec         int
-	ApplePushSettings       []ApplePushSettings
-	EnableMetrics           bool
-	EnableConsoleLog        bool
+	AndroidPushSettings  []AndroidPushSettings
+	ListenAddress        string
+	ThrottleVaryByHeader string
+	// Deprecated: Use it is maintained for backward compatibility of the Logger struct. Use LoggingCfgFile or LoggingCfgJSON instead.
+	LogFileLocation   string
+	SendTimeoutSec    int
+	RetryTimeoutSec   int
+	ApplePushSettings []ApplePushSettings
+	EnableMetrics     bool
+	// Deprecated: Same reason as LogFileLocation.
+	EnableConsoleLog bool
+	// Deprecated: Same reason as LogFileLocation.
 	EnableFileLog           bool
+	LoggingCfgFile          string
+	LoggingCfgJSON          string
```

Comment:
There is no need to expose so many complicated logging options. We just need to expose only one additional config - log in json or not. All other config options can remain as-is and do not need to be deprecated.

So therefore:
```
EnableConsoleLog
EnableFileLog
LogFileLocation
LogFormat - json or plain.
```

That's it.